### PR TITLE
#880, make the demo at /#virtual-paging work properly

### PR DESCRIPTION
**Pull request by ivankolev**
set rowCount to rows.length, to prevent 'Error: query at index -1 failed: Fenwick tree array not initialized.' and make the demo work properly at https://swimlane.github.io/ngx-datatable/#virtual-paging

**What kind of change does this PR introduce?** (check one with "x")
- [ x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
As described in the issue #880 and currently on https://swimlane.github.io/ngx-datatable/#virtual-paging or http://localhost:9999/#virtual-paging, virtual paging demo throws an error because the rowCount does not properly reflect the number of rows

**What is the new behavior?**
Fixes the said javascript error and make the demo work as intended

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x ] No

### CODE REVIEW

File: src/components/body/body.component.ts [hunk 1]
```diff
@@ -125,6 +125,7 @@ export class DataTableBodyComponent implements OnInit, OnDestroy {
   @Input() set rows(val: any[]) {
     this._rows = val;
     this.rowExpansions.clear();
+    this._rowCount = this._rows.length;
```
Comment (amcdnl) [inline src/components/body/body.component.ts]: This won't work right since the row count varies. There is a function to get this information.

**Comment by ivankolev** [comment]
Ok, back to the drawing board then. It fixes the error and all the tests passed, so I assumed it was enough. So, just to make it clear, there could be cases where rowCount is not equal to the number of rows?

**Comment by amcdnl** [comment]
Server side paging.

File: src/components/body/body.component.ts (hunk 1 above)
Comment (ivankolev) [reply src/components/body/body.component.ts]: Aah, I see that /#server-paging is indeed broken with my change. Wonder if it's worth it to maybe add more e2e tests that could potentially catch such errors.

**Comment by ivankolev** [comment]
Alright, I have pushed my second attempt at it, both /#server-paging and /#virtual-paging seem to be working now...

**Comment by Val1962** [comment]
@amcdnl Any directions if you think suggested fix is not good?
It would be good to make virtual-paging working, the first page used to load successfully. I think it stopped working after scroller performance fix https://github.com/swimlane/ngx-datatable/commit/7227e559107f531fdeaf056d9c79133dcb62fdbf to run 'scroll' listener outside of Angular.

**Pull request merged**